src/logger.service.ts

```diff
@@ -23,148 +17,141 @@ const Levels = [
-
-    if (additional === null || additional === undefined) {
-      additionalToLog = [];
-    } else {
-      additionalToLog = additional.map((val: any, idx: number) => {
-        try {
-          return val === null || val === undefined || typeof val === 'string' ? val
-            : JSON.stringify(val, null, 2);
-        } catch (e) {
-          return `The additional[${idx}] value could not be parsed using JSON.stringify().`;
-        }
-      });
-    }
-
-    this.http.post(this.options.serverLoggingUrl,
-      {
-        level: Levels[level],
-        message: messageToLog,
-        additional: additionalToLog,
-        timestamp: this._timestamp()
-      },
-      {
-        headers
-      })
-      .subscribe(
-        (res: any) => this._log(NgxLoggerLevel.TRACE, false, 'Server logging successful', [res]),
-        (error: HttpErrorResponse) => this._log(NgxLoggerLevel.ERROR, false, 'FAILED TO LOG ON SERVER', [error])
-      );
+  public updateConfig(config: LoggerConfig) {
+    this.configService.updateConfig(config);
   }
 
-  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = []): void {
+  private _logIE(level: NgxLoggerLevel, message: string, additional: any[] = [], timestamp: string): void {
     switch (level) {
       case NgxLoggerLevel.WARN:
-        console.warn(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.warn(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.ERROR:
-        console.error(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.error(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       case NgxLoggerLevel.INFO:
-        console.info(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.info(`${timestamp} [${Levels[level]}] `, message, ...additional);
         break;
       default:
-        console.log(`${this._timestamp()} [${Levels[level]}] `, message, ...additional);
+        console.log(`${timestamp} [${Levels[level]}] `, message, ...additional);
     }
   }
 
-  private _log(level: NgxLoggerLevel, logOnServer: boolean, message, additional: any[] = []): void {
-    if (!message) {
-      return;
-    }
-
+  private _prepareMessage(message) {
     try {
       if (message instanceof Error) {
         message = message.stack;
       } else if (typeof message !== 'string') {
         message = JSON.stringify(message, null, 2);
       }
     } catch (e) {
-      additional = [message, ...additional];
+      // additional = [message, ...additional];
       message = 'The provided "message" value could not be parsed with JSON.stringify().';
     }
 
-    // Allow logging on server even if client log level is off
-    if (logOnServer) {
-      this._logOnServer(level, message, additional);
+    return message;
+  }
+
+  private _prepareAdditionalParameters(additional: any[]) {
```

Comment:
The `additional` parameter should be defined like this (since you are checking for `null` / `undefined` just below:
```javascript
private _prepareAdditionalParameters(additional?: any[]) {
```

Reply:
I don't want to make the parameter optional, because I don't want this method to be called like this: `this._prepareAdditionalParameters()`. However, even though it may be called like this `this._prepareAdditionalParameters(additional)`, additional may still be null or undefined, so we should protect against it.

Reply:
@dbfannin => So what happens when the user only wants to log a message and not `additional` parameters? In that case, isn't this considered `optional` or `null` | `undefined`?

Reply:
fair point. This still doesn't mean that additional should be made optional, but it would probably be cleaner if additional were null or undefined if it always return null; The spread operator handles making sure the logger doesn't print anything if it is an empty array, but for server logging purposes, it could be confusing. I'll make some adjustments here.

`console.log(`%c${timestamp} [${logLevelString}]`, `color:${color}`, message, ...additional);`